hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -74,11 +79,13 @@ case class HoodieTableState(tablePath: String,
 abstract class HoodieBaseRelation(val sqlContext: SQLContext,
                                   val metaClient: HoodieTableMetaClient,
                                   val optParams: Map[String, String],
-                                  userSchema: Option[StructType])
+                                  userSchema: Option[StructType],
+                                  val canPruneSchema: Boolean = true)
   extends BaseRelation
     with FileRelation
     with PrunedFilteredScan
-    with Logging {
+    with Logging
+    with SparkAdapterSupport {
```

Comment:
Based on the changes, the spark.sql.extensions config must be added for spark-shell in all spark versions.  We should update the docs as well.